From 7fe5b592ec024870f1bf1ec56069f4f36b0460f7 Mon Sep 17 00:00:00 2001
From: Art Limbaga <limbaga@ualberta.ca>
Date: Wed, 22 Mar 2023 21:17:27 -0600
Subject: [PATCH 02/13] shiftview and shifttag

---
 config.def.h |  4 ++++
 dwm.c        | 29 +++++++++++++++++++++++++++++
 2 files changed, 33 insertions(+)

diff --git a/config.def.h b/config.def.h
index c225943..8c99ad5 100644
--- a/config.def.h
+++ b/config.def.h
@@ -82,6 +82,10 @@ static const Key keys[] = {
 	{ MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
 	{ MODKEY,                       XK_Left,   focusmon,       {.i = -1 } },
 	{ MODKEY,                       XK_Right,  focusmon,       {.i = +1 } },
+	{ MODKEY,                       XK_g,      shiftview,      { .i = -1 } },
+	{ MODKEY|ShiftMask,             XK_g,      shifttag,       { .i = -1 } },
+	{ MODKEY,                       XK_semicolon,	shiftview,   { .i = +1 } },
+	{ MODKEY|ShiftMask,             XK_semicolon,	shifttag,    { .i = +1 } },
 	{ MODKEY|ShiftMask,             XK_Left,   tagmon,         {.i = -1 } },
 	{ MODKEY|ShiftMask,             XK_Right,  tagmon,         {.i = +1 } },
 	TAGKEYS(                        XK_1,                      0)
diff --git a/dwm.c b/dwm.c
index c2bd871..1f49839 100644
--- a/dwm.c
+++ b/dwm.c
@@ -204,6 +204,8 @@ static void setlayout(const Arg *arg);
 static void setmfact(const Arg *arg);
 static void setup(void);
 static void seturgent(Client *c, int urg);
+static void shifttag(const Arg *arg);
+static void shiftview(const Arg *arg);
 static void showhide(Client *c);
 static void spawn(const Arg *arg);
 static void tag(const Arg *arg);
@@ -1626,6 +1628,33 @@ seturgent(Client *c, int urg)
 	XFree(wmh);
 }
 
+void
+shifttag(const Arg *arg)
+{
+	Arg shifted;
+	shifted.ui = selmon->tagset[selmon->seltags];
+
+
+	if (arg->i > 0)	/* left circular shift */
+		shifted.ui = ((shifted.ui << arg->i) | (shifted.ui >> (LENGTH(tags) - arg->i)));
+	else		/* right circular shift */
+		shifted.ui = (shifted.ui >> (- arg->i) | shifted.ui << (LENGTH(tags) + arg->i));
+	tag(&shifted);
+}
+
+void
+shiftview(const Arg *arg)
+{
+	Arg shifted;
+	shifted.ui = selmon->tagset[selmon->seltags];
+
+	if (arg->i > 0)	/* left circular shift */
+		shifted.ui = (shifted.ui << arg->i) | (shifted.ui >> (LENGTH(tags) - arg->i));
+	else		/* right circular shift */
+		shifted.ui = (shifted.ui >> (- arg->i) | shifted.ui << (LENGTH(tags) + arg->i));
+	view(&shifted);
+}
+
 void
 showhide(Client *c)
 {
-- 
2.40.0

